### PR TITLE
Allow ffmpeg codec to be set via config

### PR DESCRIPTION
**Pull request by brianmario**
This refectors the OMX stuff out to instead just allow passing the `ffmpeg` codec via config value. This means a user can use whatever hardware acceleration codec is supported on the hardware they're using. For myself, I'm running on a Mac Mini so I want to use the `videotoolbox` codecs.

This makes a couple of other changes like the default bitrate from 300 (bytes?) to 1,500,000 (1.5Mbit - which can be specified as "1500k" with `ffmpeg`). It also gets rid of the `fps` value since ffmpeg will automatically detect and passthrough the fps. Lastly, it cleans up some extra whitespace and adds newlines to files that were missing it. Any or all of these could be reverted if you'd like. The main thing I'm going for here is allowing the codec to be set via config.

### CODE REVIEW

File: lib/nestcam.js [hunk 1]
```diff
@@ -22,14 +22,14 @@ class NestCam {
...
-  configureWithHAP(hap, useOMX) {
+  configureWithHAP(hap, config) {
     uuid = hap.uuid;
     Service = hap.Service;
     Characteristic = hap.Characteristic;
     StreamController = hap.StreamController;
 
     let self = this;
-    self.useOMX = useOMX;
+    self.ffmpegCodec = config.ffmpegCodec;
```
Comment (KhaosT) [inline lib/nestcam.js]: if you want to do this, let's have a fallback to default `libx264` (or omx if legacy useOMX flag is true)

File: lib/streamer.js [hunk 2]
```diff
@@ -406,19 +406,13 @@ class NexusStreamer extends EventEmitter {
     if (sessionInfo) {
       let width = 1280;
       let height = 720;
-      let fps = 30;
-      let bitrate = 300;
+      let bitrate = "1500k";
```
Comment (KhaosT) [inline lib/streamer.js]: original bitrate is `300k`. HomeKit is really sensitive about this and fps, especially during remote access.

Comment (KhaosT) [reply lib/streamer.js]: And there is already a "k" in the ffmpeg command...

Comment (brianmario) [reply lib/streamer.js]: Oh ok interesting...

Sorry I missed that in the original cmd string string 🤦‍♂️

File: lib/nestcam.js (hunk 1 above)
Comment (brianmario) [reply lib/nestcam.js]: I did in one place but missed it here. Will update.

This replaces the useOMX flag entirely, so I'll add a check if it's set in the config and set the codec appropriately.

File: lib/nestcam.js [hunk 3]
```diff
@@ -22,14 +22,19 @@ class NestCam {
...
     Service = hap.Service;
     Characteristic = hap.Characteristic;
     StreamController = hap.StreamController;
 
     let self = this;
-    self.useOMX = useOMX;
+    // This is for backward compatibility with the old useOMX config value
+    if (config.useOMX) {
+      self.ffmpegCodec = "omx";
+    } else if (config.ffmpegCodec) {
```
[the remark this answers is not in the document]
Comment (KhaosT) [reply lib/nestcam.js]: This should be h264_omx?

File: lib/streamer.js [hunk 4]
```diff
@@ -15,12 +15,12 @@ const StartPlayback = require('./StartPlayback.js').StartPlayback;
...
 
 class NexusStreamer extends EventEmitter {
-  constructor(host, cameraUUID, sessionToken, useOMX) {
+  constructor(host, cameraUUID, sessionToken, config) {
     super();
     let self = this;
     self.isStreaming = false;
     self.authorized = false;
-    self.useOMX = useOMX;
+    self.ffmpegCodec = config.ffmpegCodec;
```
Comment (KhaosT) [inline lib/streamer.js]: Here you are accessing `config.ffmpegCodec` but in `NestCam.configureWithHAP` we are doing the fallback on `NestCam.ffmpegCodec`, maybe either have `NestCam.configureWithHAP` override config directly, or pass the `ffmpegCodec` as part of the `NexusStreamer` constructor?

**Comment by KhaosT** [comment]
Awesome :) Thanks for doing this 👍

**Comment by brianmario** [comment]
Sorry for the back and forth on that diff. I should have probably waited until I wasn't sick in bed to do this. Also appreciate the help (and patience!). OSS is sorely in need of more of it, so thank you.

**Pull request merged**